systems/notification/mailer/pkg/server/mailer.go

```diff
@@ -22,139 +21,165 @@ import (
+	}
+	defer c.Close()
+	config := &tls.Config{
+		ServerName: s.mailer.Host,
 	}
+	if err = c.StartTLS(config); err != nil {
+		return nil, status.Error(codes.Internal, "failed to start TLS")
+	}
+
+	auth := smtp.PlainAuth("", s.mailer.Username, s.mailer.Password, s.mailer.Host)
 
-	var recipientList []string
-	if len(to) > 0 {
-		recipientList = make([]string, len(to))
-		copy(recipientList, to)
+	// Authenticate with the SMTP server
+	if err = c.Auth(auth); err != nil {
+		return nil, status.Error(codes.Internal, "failed to authenticate with SMTP server")
 	}
 
-	msg := "From: " + from + "\r\n" +
-		"To: " + strings.Join(recipientList, ",") + "\r\n" +
-		"Subject: " + subject + "\r\n" +
-		"MIME-Version: 1.0\r\n" +
-		"Content-Type: text/html; charset=utf-8\r\n" +
-		"\r\n" +
-		bodyBuffer.String()
+	// Set the sender email address
+	if err = c.Mail(s.mailer.From); err != nil {
+		return nil, status.Error(codes.Internal, "failed to set sender email address")
+	}
 
-	auth := smtp.PlainAuth("", s.mailer.Username, s.mailer.Password, s.mailer.Host)
-	port := strconv.Itoa(s.mailer.Port)
-	err = smtp.SendMail(s.mailer.Host+":"+port, auth, from, recipientList, []byte(msg))
+	// Add the recipient email addresses
+	for _, recipient := range req.To {
+		if err = c.Rcpt(recipient); err != nil {
+			return nil, status.Error(codes.Internal, "failed to add recipient email address")
+		}
+	}
+
+	w, err := c.Data()
 	if err != nil {
-		log.Errorf("Failed to send email: %v", err.Error())
-		err = s.mailerRepoRepo.SendEmail(&db.Mailing{
-			MailId:  mailId,
-			Email:   recipientList[0], // Use the first email if only one is provided
-			Subject: subject,
-			Body:    bodyBuffer.String(),
-			SentAt:  sentAt,
-			Status:  "failed",
-		})
-		return nil, err
+		return nil, status.Error(codes.Internal, "failed to open data connection")
 	}
+	defer w.Close()
 
-	log.Infof("Email sent successfully to %v", recipientList)
+	_, err = w.Write(body.Bytes())
+	if err != nil {
+		return nil, status.Error(codes.Internal, "failed to write email body")
+	}
 
-	for _, recipient := range recipientList {
-		err = s.mailerRepoRepo.SendEmail(&db.Mailing{
-			MailId:  mailId,
-			Email:   recipient,
-			Subject: subject,
-			Body:    bodyBuffer.String(),
-			SentAt:  sentAt,
-			Status:  "sent",
+	err = c.Quit()
+	if err != nil {
+		log.Errorln("Email not sent ", err.Error())
+		err = s.mailerRepo.SendEmail(&db.Mailing{
+
+			MailId:       mailId,
+			Email:        req.To[0],
+			TemplateName: req.TemplateName,
+			Status:       "failed sent",
 		})
 		if err != nil {
-			log.Error("Error while sending email" + err.Error())
-			return nil, grpc.SqlErrorToGrpc(err, "Failed to send email")
+			log.Error("Error while saving email" + err.Error())
+			return nil, grpc.SqlErrorToGrpc(err, "failed to get email")
 		}
 	}
 
-	response := &pb.SendEmailResponse{
-		Message: "Email sent successfully",
+	err = s.mailerRepo.SendEmail(&db.Mailing{
```

Comment:
either way, we still store the email into DB.

Reply:
Yes we do still sent because we have a status column.